File: online-judge/CodeForces/CodeForces 1189D1.py
n = int(input())
cnt = [0] * (n + 1)
for i in range(n - 1):
    u, v = map(int, input().split())
    cnt[u] += 1
    cnt[v] += 1
print('NO' if 2 in cnt else 'YES')

File: online-judge/CodeForces/CodeForces 1189D1.cpp
#include <iostream>
#include <algorithm>

using namespace std;

int cnt[100007];

int main()
{
    int n;
    cin >> n;
    for (int i = 0; i < n - 1; i++)
    {
        int u, v;
        cin >> u >> v;
        cnt[u]++;
        cnt[v]++;
    }
    if (find(cnt + 1, cnt + n + 1, 2) == cnt + n + 1)
    {
        cout << "YES" << endl;
    }
    else
    {
        cout << "NO" << endl;
    }
    return 0;
}